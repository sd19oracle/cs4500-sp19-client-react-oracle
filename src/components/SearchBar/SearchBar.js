import React from 'react'

const SearchBar = ({history}) =>
    <div className="input-group input-group-lg">
        <input
            placeholder="Search for providers"
            type="text"
            className="form-control"/>
        <input
            placeholder="Zip code"
            type="text"
            className="form-control"/>
        <div className="input-group-append">
            <button
                onClick={() => {
                    history.push('/providers')
                }}
                className="btn btn-primary"
                type="button">
                Search
            </button>
        </div>
    </div>

export default SearchBar