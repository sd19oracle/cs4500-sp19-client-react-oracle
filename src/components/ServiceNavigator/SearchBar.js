import React from "react";

const SearchBar = ({handler, placeholder}) => (
  <input
    placeholder={placeholder}
    className={"form-control ml-1"}
    onChange={handler}
  />
);
export default SearchBar;
