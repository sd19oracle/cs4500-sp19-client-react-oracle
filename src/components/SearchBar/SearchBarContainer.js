import React from 'react'
import SearchBarService from '../../services/SearchBarService'
import SearchBar from './SearchBar';

class SearchBarContainer extends React.Component {
    constructor(props) {
        super(props)
        this.searchBarService = SearchBarService.getInstance()
        this.state = {
            providersList: [],
            name: "",
            zip: ""

        }
        this.searchProvByName = this.searchProvByName.bind(this)
        this.searchProvByZip = this.searchProvByZip.bind(this)
        this.searchProvByNameAndZip = this.searchProvByNameAndZip.bind(this)
        this.pressSearch = this.pressSearch.bind(this)
    }

    searchProvByName(name) {
        this.searchBarService.findProvidersByName(name)
            .then(providers =>
                this.setState({
                    providersList: providers
                })
            )
    }
    searchProvByZip(zip) {
        this.searchBarService.findProvidersByZip(zip)
            .then(providers =>
                this.setState({
                    providersList: providers
                })
            )
    }
    searchProvByNameAndZip(name, zip) {
        this.searchBarService.findProvidersByNameAndZip(name, zip)
            .then(providers =>
                this.setState({
                    providersList: providers
                })
            )
    }

    pressSearch() {
        this.props.history.push('')
    }

    updateName = e => {
        this.setState({
            name: e.target.value
        });
    };

    updateZip = e => {
        this.setState({
            zip: e.target.value
        })
    }

    render() {
        return (
            <div>
                <SearchBar
                    updateName={this.updateName}
                    updateZip={this.updateZip}
                    name={this.state.name}
                    zip={this.state.zip}
                    pressSearch={this.pressSearch}
                />
            </div>
        )
    }
}

export default SearchBarContainer