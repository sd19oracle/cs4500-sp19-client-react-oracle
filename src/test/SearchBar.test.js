import React from 'react'
import TestRenderer from 'react-test-renderer';
import SearchBar from '../components/SearchBar/SearchBar'
import providerList from '../components/ProvidersPage/ServiceProviderList'
import SearchBarComponent from '../components/SearchBar/SearchBarContainer'
import ProviderList from '../data/provider_list.mock.json'

// Snapshot testing
test('basic test', () => {
    const testRenderer = TestRenderer.create(
        <SearchBar name=""
                    zip="0"/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
});


// Snapshot testing
test('basic test2', () => {
    const testRenderer = TestRenderer.create(
        <SearchBar name=""
                    zip="02115"/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
});


// Snapshot testing
test('basic test3', () => {
    const testRenderer = TestRenderer.create(
        <SearchBar name="jordan"
                    zip="02115"/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
});


// DOM testing
test('rendering results on provders page', () => {
    const testRenderer = TestRenderer.create(
        <providerList providerList={ProviderList}/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
});