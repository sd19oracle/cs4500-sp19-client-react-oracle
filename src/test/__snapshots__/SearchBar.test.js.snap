// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test 1`] = `
<div
  className="input-group input-group-lg"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search for providers"
    type="text"
    value=""
  />
  <input
    className="form-control"
    maxLength={5}
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Zip code"
    type="text"
    value="0"
  />
  <div
    className="input-group-append"
  >
    <button
      className="btn btn-primary"
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;

exports[`basic test2 1`] = `
<div
  className="input-group input-group-lg"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search for providers"
    type="text"
    value=""
  />
  <input
    className="form-control"
    maxLength={5}
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Zip code"
    type="text"
    value="02115"
  />
  <div
    className="input-group-append"
  >
    <button
      className="btn btn-primary"
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;

exports[`basic test3 1`] = `
<div
  className="input-group input-group-lg"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Search for providers"
    type="text"
    value="jordan"
  />
  <input
    className="form-control"
    maxLength={5}
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Zip code"
    type="text"
    value="02115"
  />
  <div
    className="input-group-append"
  >
    <button
      className="btn btn-primary"
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;

exports[`rendering results on provders page 1`] = `
<providerList
  providerList={
    Array [
      Object {
        "answers": Array [],
        "email": "ana@gmail.com",
        "firstName": "Ana",
        "id": 890,
        "lastName": "Christina",
        "password": "ana",
        "role": "provider",
        "services": Array [],
        "zipCode": "02110",
      },
      Object {
        "answers": Array [],
        "email": "song@gmail.com",
        "firstName": "Song",
        "id": 789,
        "lastName": "Park",
        "password": "itisfine",
        "role": "provider",
        "services": Array [],
        "zipCode": "02115",
      },
      Object {
        "answers": Array [],
        "email": "jordan@gmail.com",
        "firstName": "Michael",
        "id": 456,
        "lastName": "Jordan",
        "password": "goat",
        "role": "provider",
        "services": Array [],
        "zipCode": "10010",
      },
      Object {
        "answers": Array [],
        "email": "pokemon@gmail.com",
        "firstName": "Pokemon",
        "id": 567,
        "lastName": "Go",
        "password": "pokemon",
        "role": "provider",
        "services": Array [],
        "zipCode": "31256",
      },
    ]
  }
/>
`;
